### PR TITLE
Adds a xenomorph crate and pre-larval scanning experiment + alien acid glands release acid on death

### PR DESCRIPTION
**Pull request by Lufferly**
## About The Pull Request

Added the genetically modified xenomorph crate to cargo, unlocked for purchase after science researches xenobiology. The facehugger inside can create a xenomorph, but said xenomorph will be infertile and only be able to create sterile facehuggers. Infertile queens do not count towards the queen cap, but will still weaken the hive when they die.
As incentive for science to buy and raise a xenomorph, added an experiment for scanning pre-larval xenomorph lifeforms (eggs, facehuggers, and embryos), unlocked for experimentation after researching xenobiology. Scanning enough grants a hefty 500 research points.
Additionally, alien acid glands now release acid on death, which has a side affect of destroying all of the holding mobs internal organs (besides the acid gland).

## Why It's Good For The Game

Main inspiration was to add some department interaction between cargo and science, and add a slightly dangerous but rewarding project for the entire science department. If science wants the points, or just the challenge, they can try their hand at setting up containment for a xenomorph. If they fail its bad, but not round ending since an escaped xenomorph wont go on to create an army. Either way its bound to add some tension to the shift.
Alien acid glands releasing acid on death is interesting, and also means it will be harder for the crew to harvest organs from the xenos they created, as thats not something im too interested in seeing exploited. Of course, hunters dont have acid glands, so if the alien chooses that its their own fault.

## Changelog

:cl: Seven
add: You can now buy a genetically modified facehugger from cargo, unlocked after science researched xenobiology.
add: Added a science experiment for scanning pre-larval xenomorph lifeforms.
add: Xeno acid glands now release acid on death and destroy all other internal organs besides itself.
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
this sounds really cool, but what if you could operate on a xeno to make it sterile?

**Comment by CRITAWAKETS** [comment]
Can you still take the queen's organs if you do this? If so, powergamers are veeery likely to start raising these xenos in 1x1 reinforced wall boxes in order to get the organs, leading to the most boring gameplay ever. This already sometimes happens with the random xenobio egg spawn, if the scientist actually wants to do this.

**Comment by Iamgoofball** [comment]
absolutely not holy shit

**Comment by carlarctg** [comment]
they spaye the god damnf acehugger

**Comment by nianjiilical** [comment]
the funny thing is, without the ability to reproduce i dont think most players would even go queen

if you're going to be the only xeno anyway just go hunter or *maybe*p praetorian

**Comment by Lufferly** [comment]
> Can you still take the queen's organs if you do this? If so, powergamers are veeery likely to start raising these xenos in 1x1 reinforced wall boxes in order to get the organs, leading to the most boring gameplay ever. This already sometimes happens with the random xenobio egg spawn, if the scientist actually wants to do this.

Cant think of a clean way to counter this. If this is isnt already against the rules and likely to be a problem then this probably shouldnt get merged.

**Comment by MrMelbert** [comment]
Looking at the code, you tied the sterility to the organs, so that should mean people can't kill the queen to raise an army, right?

**Comment by Lufferly** [comment]
> Looking at the code, you tied the sterility to the organs, so that should mean people can't kill the queen to raise an army, right?

Correct, the queens will get a sterile eggsac which can only create sterile facehuggers.

**Comment by zoomachina** [comment]
this sounds good but
> unlocked for purchase after science researches xenobiology

why this restriction?
> will still weaken the hive when they die

that hive would apparently be produced by some other queen, so why should this happen?

**Comment by MrMelbert** [comment]
On one hand I think this is cool because I think science experimenting on aliens (not necessarily just xenos and slimes mind you, but aliens in general) is cool, so encouraging it = good

On the other hand it'll make the 1% egg spawn less special which is sad

**Comment by SomeoneYouProbablyKnow** [comment]
There should be a way to make the xeno not-sterlie. Clearly it should be an emag_act.

**Comment by SimplySmiley123** [comment]
> > Can you still take the queen's organs if you do this? If so, powergamers are veeery likely to start raising these xenos in 1x1 reinforced wall boxes in order to get the organs, leading to the most boring gameplay ever. This already sometimes happens with the random xenobio egg spawn, if the scientist actually wants to do this.
> 
> Cant think of a clean way to counter this. If this is isnt already against the rules and likely to be a problem then this probably shouldnt get merged.

I mean, the egg laying gland could just be making sterlie facehugger eggs instead of real ones and a majority of crew going after the xeno organs would miraculously not care anymore, would just be the ones looking for the xeno eyes left.

**Comment by SimplySmiley123** [comment]
> There should be a way to make the xeno not-sterlie. Clearly it should be an emag_act.

How
       how do you swipe an emag on a xenomorph

**Comment by SomeoneYouProbablyKnow** [comment]
Yes, we need more emag acts, double the already insane amount we already have. You should be able to emag every single items ever

**Comment by Exester509** [comment]
> > Can you still take the queen's organs if you do this? If so, powergamers are veeery likely to start raising these xenos in 1x1 reinforced wall boxes in order to get the organs, leading to the most boring gameplay ever. This already sometimes happens with the random xenobio egg spawn, if the scientist actually wants to do this.
> 
> Cant think of a clean way to counter this. If this is isnt already against the rules and likely to be a problem then this probably shouldnt get merged.

It's not really against the rules (unless you're that one person spamming it every single round) and it's definetely going to be abused. Best thing you can do is make the organs melt when the xeno dies because a flaw in their new genetics floods them with potent acid when they die or whatever. It has to be balanced, not explainable

**Comment by optimumtact** [comment]
> There should be a way to make the xeno not-sterlie. Clearly it should be an emag_act.

Stop trolling

**Comment by Iamgoofball** [comment]
@optimumtact anomaly core

**Comment by LSommers** [comment]
I mean, if you gave it a variant organ to solve _one_ issue, you could always like.
Just.
Do it again? In a way that makes anyone trying to organ harvest cry when they realize all the organs are actually terrible and the RD is now going to beat them to death for killing their specimen?

**Comment by Lufferly** [comment]
Alien acid glands now release acid and destroy the internal organs of the holding mob, on all xenos (not just bought ones). How the crew is supposed to harvest the queens organs now is left as an exercise for the reader.

**Comment by norsvenska** [comment]
and so the domestication of the xenomorph begins

**Comment by shellspeed1** [comment]
Not sure I'm a big fan. This PR is massively affecting how a crew can respond to xenomorph infestations (Using hive nodes typically to see if any are still alive). This can make it very difficult for crews to catch xenomorphs that are out on the prowl  in order to figure out where a hive might be and monitor for the existence of further xenomorphs.

Another point, this is effectively an antag that can be purchased by anyone looking to cause grief or a distraction for 10k. All you need to get the xenomorph once you have the crate is a monkey which can be done with relative ease. There isn't even really a chance of the xenomorph turning on you as you can just walk away and let it do its thing. Boom, that is an antag with vent crawl that can and will do stuff like destroying the ore silo, telecomms, and becoming the entire focus of the security department until people get used to it in which case it makes actual xenomorph hives absolute round enders.

The last issue I have with this PR is that its not going to exactly add the tension that it is advertising as the experiments focus on larval xenos (Huggers and embryos) instead of fully grown ones. There is zero containment risk here as you can once again just let a monkey get infected and just rip it out. If you want an actual containment risk, you should aim for xenomorphs that are more evolved for the scans.

On one last tidbit, 99% of the work for xenomorph containment is already done for you in xenobiology. This isn't exactly hard to do if you aren't eating crayons like the marines normally sent to deal with these issues.

**Comment by ATH1909** [comment]
> Alien acid glands now release acid and destroy the internal organs of the holding mob on death, on all xenos (not just bought ones). How the crew is supposed to harvest the queens organs now is left as an exercise for the reader.

Not a fan of this happening for all xenos, I liked being able to borg them and create human-queen hybrids.

**Comment by Addust** [comment]
imo:
- organ melt should only apply to neutered xenos (dna fuckery and what have you) instead of wild ones so if you kill an actual queen you get to do organ fuckery
- the hive node should be a domestic hive node that communicates on a different channel so people cant just instantly meta "ooga unga xeno base in viro" every time

**Comment by Exester509** [comment]
> imo:
> 
> * organ melt should only apply to neutered xenos (dna fuckery and what have you) instead of wild ones so if you kill an actual queen you get to do organ fuckery
> * the hive node should be a domestic hive node that communicates on a different channel so people cant just instantly meta "ooga unga xeno base in viro" every time

Agreed. Melting actual antag xeno organs isn't a good thing because to get them a xeno infestation has to roll, you gotta kill a xeno, and then harvest its organs. It's a much less controlled environment than xenobio being observed by five scientists and a secoff with a laser.

File: code/game/objects/structures/aliens.dm [hunk 1]
```diff
@@ -344,12 +344,16 @@
...
+	/// If true, the hugger inside will be sterile
+	var/sterile = FALSE
 
 /obj/structure/alien/egg/Initialize(mapload)
 	. = ..()
 	update_appearance()
 	if(status == GROWING || status == GROWN)
 		child = new(src)
+		if (sterile == TRUE)
+			child.sterile = TRUE
```
Comment (Jacquerel) [inline code/game/objects/structures/aliens.dm:356]: Can just do 
```suggestion
		child.sterile = sterile
```

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 2]
```diff
@@ -127,6 +130,9 @@ Des: Removes all infected images from the alien.
 	new_xeno.setDir(dir)
 	new_xeno.change_name(name, real_name, numba)
 
+	if (infertile == TRUE)
+		new_xeno.infertile = TRUE
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/alien/alien.dm:134]:
```suggestion
	new_xeno.infertile = infertile
```

File: code/modules/mob/living/carbon/alien/adult/caste/praetorian.dm [hunk 3]
```diff
@@ -52,5 +52,13 @@
 /datum/action/cooldown/alien/evolve_to_queen/Activate(atom/target)
 	var/mob/living/carbon/alien/adult/royal/evolver = owner
 	var/mob/living/carbon/alien/adult/royal/queen/new_queen = new(owner.loc)
+	// If we are infertile, replace the queens eggsac with an eggsac that lays sterile eggs
+	if (evolver.infertile == TRUE)
+		var/obj/item/organ/internal/alien/eggsac/eggsac = new_queen.get_organ_slot(ORGAN_SLOT_XENO_EGGSAC)
+		qdel(eggsac)
+		var/obj/item/organ/internal/alien/eggsac/sterile/new_eggsac = new()
+		new_queen.organs += new_eggsac
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/alien/adult/caste/praetorian.dm:60]: Does `Insert` not add it to this list automatically?

File: code/modules/mob/living/carbon/alien/organs.dm [hunk 4]
```diff
@@ -158,7 +158,37 @@
...
+	releasing_acid.my_atom = owner
+	releasing_acid.create_foam(/datum/effect_system/fluid_spread/foam, acid_release_on_death)
+
+	// Destroy all of the holders internal organs
+	for (var/obj/item/organ/internal/organ as anything in owner.organs)
+		if (organ == src)
+			continue
+		if (istype(organ, /obj/item/organ/internal/brain))
+			continue
+		organ.Remove(owner)
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/alien/organs.dm:190]: Can organs have item flags which make them acidproof? Might be worth checking against those (and then applying them to this organ so you can skip the src check)

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 5]
```diff
@@ -123,6 +125,9 @@
 	SEND_SOUND(new_xeno, sound('sound/voice/hiss5.ogg',0,0,0,100)) //To get the player's attention
 	new_xeno.add_traits(list(TRAIT_HANDS_BLOCKED, TRAIT_IMMOBILIZED, TRAIT_NO_TRANSFORM), type) //so we don't move during the bursting animation
 	new_xeno.SetInvisibility(INVISIBILITY_MAXIMUM, id=type)
+	// If we are genetically modified, future xenos of this lineage will only be able to create sterile huggers
+	if (genetically_modified == TRUE)
+		new_xeno.infertile = TRUE
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:130]:
```suggestion
	new_xeno.infertile = genetically_modified
```

File: code/modules/mob/living/carbon/alien/special/facehugger.dm [hunk 6]
```diff
@@ -215,7 +218,10 @@
 
 		var/obj/item/bodypart/chest/LC = target.get_bodypart(BODY_ZONE_CHEST)
 		if((!LC || IS_ORGANIC_LIMB(LC)) && !target.get_organ_by_type(/obj/item/organ/internal/body_egg/alien_embryo))
-			new /obj/item/organ/internal/body_egg/alien_embryo(target)
+			var/obj/item/organ/internal/body_egg/alien_embryo/embryo = new(target)
+			if (genetically_modified == TRUE)
+				embryo.genetically_modified = TRUE
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/alien/special/facehugger.dm:223]:
```suggestion
			embryo.genetically_modified = TRUE
```

File: code/modules/mob/living/carbon/alien/organs.dm (hunk 4 above)
Comment (Jacquerel) [reply code/modules/mob/living/carbon/alien/organs.dm:190]: Also TBH I think being able to remove xeno organs from most xenos is like... intentional.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Code notes are mostly just nitpicks. Sorry this has been sat so long without any attention.

I am a _little_ wary of this idea but willing to give it a go, but there are some things I would like change or reassurance on first:

Firstly, I think melting all xeno organs including from other xenos is a little sad, getting random people implanted with xeno organs is fun and would be a shame to lose.
I understand the concern about people using this to farm xeno organs, but (correct me if I am wrong) I think that would usually be _mostly_ fine when they can't lay eggs? 
Alternately go with someone else's suggestion and give them a full variant organ set I guess.

Secondly, just to be sure... is it clear to a player being offered this role that they are being offered the role of a xenomorph guinea pig which cannot lay eggs? It's not going to trick people into thinking they're going to get to play as a real xeno right?
To be honest it might even be better to have these facehuggers produce NPC xenomorphs but it depends if people actually enjoy "captive alien" RP more than I think they do.

Thirdly, maybe the one you might dislike most... I don't love that this is a queen and the side effects of it being a queen hurting an actual hive in the round. I don't know if this would ruin the fun for you but making it so that this larvae can only become a sentinel or hunter might fix some issues and still be neat for R&D players. I also suspect (though no promises) that admins are much more likely to be forgiving of people letting their alien puppy out of its cage to see what happens if it's a sentinel rather than a queen.

**Comment by Lufferly** [comment]
I dont got time for this rn, someone can pick it up or I may come back to it later

**Pull request closed without merging**